src/scoring/hpf_scoring.py

```diff
@@ -159,130 +113,100 @@ def predict(self, input_stack):
             return [], {}, list(missing_packages)
         manifest_match = self.match_manifest(input_id_set)
         if manifest_match > 0:
-            companion_recommendation = self.recommend_known_user(manifest_match, input_id_set)
+            result, user_id = self.recommend_known_user(manifest_match)
         else:
-            companion_recommendation = self.recommend_new_user(list(input_id_set))
+            try:
+                result, user_id = self.recommend_new_user(list(input_id_set))
+                companion_recommendation = self.filter_recommendation(result, input_id_set, user_id)
+            except ValueError as e:
+                _logger.error(e)
+                return companion_recommendation, package_topic_dict, list(missing_packages)
         return companion_recommendation, package_topic_dict, list(missing_packages)
 
-    def match_feedback_manifest(self, input_id_set):
-        """Find a feedback manifest that matches user's input package list and return its index.
-
-        :param input_id_set: A set containing package ids of user's input package list.
-        :return manifest_id: The index of the matched feedback manifest.
-        """
-        for manifest_id, dependency_set in self.feedback_id_dict.items():
-            if dependency_set == input_id_set:
-                break
-        else:
-            manifest_id = -1
-        _logger.debug(
-                "input_id_set {} and feedback_manifest_id {}".format(input_id_set, manifest_id))
-        return int(manifest_id)
-
     def match_manifest(self, input_id_set):  # pragma: no cover
         """Find a manifest list that matches user's input package list and return its index.
 
         :param input_id_set: A set containing package ids of user's input package list.
         :return manifest_id: The index of the matched manifest.
         """
+        # no blank line allowed here
+
         # TODO: Go back to the difference based logic, this simpler logic will do for now.
```

Comment:
It's time to implement this TODO. We need that for good recommendations. Please check how it's done in pypi-insights, you will get an idea.